topo/node/cptx/cptx_test.go

```diff
@@ -139,6 +140,54 @@ func TestConfigPush(t *testing.T) {
 	}
 }
 
+func TestCustomPrivilegeLevel(t *testing.T) {
+	privilegePromptMap := map[string]string{
+		"exec":          "root@%s>",
+		"configuration": "root@%s#",
+	}
+
+	tests := []struct {
+		desc        string
+		hostname    string
+		shouldMatch bool
+	}{
+		{
+			desc:        "basic case",
+			hostname:    "testexample",
+			shouldMatch: true,
+		},
+		{
+			desc:        "hostname with '.'",
+			hostname:    "test.example",
+			shouldMatch: true,
+		},
+		{
+			desc:        "failure",
+			hostname:    "Test^Example",
+			shouldMatch: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			for privilegeName, prompt := range privilegePromptMap {
+				fullPrompt := fmt.Sprintf(prompt, tt.hostname)
+				privLevel, ok := customPrivLevels[privilegeName]
+				if !ok {
+					t.Fatalf("privilege %q not defined in custom privilege", privilegeName)
+				}
+				match, err := regexp.Match(privLevel.Pattern, []byte(fullPrompt))
```

Comment:
did you forget to change it to regexp.MatchString ? Or does that not work

Otherwise LGTM

Reply:
@alexmasi  I simply forgot to change this. Could you give me one more LGTM?